syntax = "proto3";

package fixbug;

// 定义下面选项，表明生成service服务类和rpc方法，默认不生成
option cc_generic_services = true;

message Resultcode
{
    int32 errcode = 1;
    bytes errmsg = 2;
}

message LoginRequest
{
    bytes name = 1;
    bytes pwd = 2;
}

message LoginResponse
{
    Resultcode result = 1;
    bool success = 3;
}

message RegisterRequest
{
    uint32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
}

message RegisterResponse
{
    Resultcode result = 1;
    bool success = 3;
}


service UserServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
     
}